src/app/services/pokemon.service.ts

```diff
@@ -27,7 +50,22 @@ export class PokemonService {
    * @returns { Observable<any> }
    * 
    */
-  getPokemons() : Observable<any> {
-    return this.http.get(`${APIURL}`)
+  getPokemons() : Observable<Pokemon[]> {
+    //@TODO: Provide offset for pagination
+    const response = this.http.get<any>(`${APIURL}?limit=${DEFAULTLIMIT}`);
+    return response.pipe(
+      map((data: any) => {
+        return data.results.map((pokemon: any) => {
+          return pokemon.name;
+        });
+      }),
+      switchMap((pokemons: string[]) => {
+        const completePokemonData : Observable<Pokemon>[] = []; 
+        pokemons.forEach((pokemon: string) => {
+          completePokemonData.push(this.getPokemon(pokemon));
+        });
+        return combineLatest(completePokemonData);
+      }),
+    ) as Observable<Pokemon[]>;
```

Comment:
```suggestion
    return this.http.get<any>(`${APIURL}?limit=${DEFAULTLIMIT}`).pipe(
      map((data: any) => data.results.map((p: any) => p.name)),
      switchMap((pokemonsNames: string[]) => {
        const completePokemonData = pokemonsNames.map(pName => this.getPokemon(pName));
        return combineLatest(completePokemonData);
      }),
    ) as Observable<Pokemon[]>;
```
oppure
```suggestion
    return this.http.get<any>(`${APIURL}?limit=${DEFAULTLIMIT}`).pipe(
      map((data: any) => data.results.map((p: any) => p.name)),
      switchMap((pokemonsNames: string[]) => combineLatest(pokemonsNames.map(pName => this.getPokemon(pName)))),
    ) as Observable<Pokemon[]>;
```

Reply:
comunque quando puoi evita `as type..` oppure di scrivere esplicitamente i tipi nelle variabili quando già c'è l'inferenza (tranne per le funzioni), ma in un `.map()` è chiaro il tipo, scrivere `: any` non serve a niente dentro un map imho